code/game/objects/items/weapons/tools.dm

```diff
@@ -497,28 +498,72 @@
 				spawn(100)
 					user.disabilities &= ~NEARSIGHTED
 
+/obj/item/weapon/weldingtool/update_icon()
+	..()
+	if(src.welding)
+		src.icon_state = "welder1"
+	else
+		src.icon_state = "welder"
+
 /obj/item/weapon/weldingtool/empty
 	start_fueled = 0
 
 /obj/item/weapon/weldingtool/largetank
 	name = "Industrial Welding Tool"
+	desc = "The cutting edge between portability and tank size."
+	icon_state = "welder_large"
 	max_fuel = 40
 	starting_materials = list(MAT_IRON = 70, MAT_GLASS = 60)
 	origin_tech = Tc_ENGINEERING + "=2"
 
+/obj/item/weapon/weldingtool/largetank/update_icon()
+	..()
+	if(src.welding)
+		src.icon_state = "welder_large1"
+	else
+		src.icon_state = "welder_large"
```

Comment:
there's no need for this because it's inherited, also no need for src

also you can make update_icon more elegant by reducing the entire proc to
`..()
icon_state = "[initial(icon_state)][if(welding) ? "1" : "0"]`